app/ExposureHistoryContext.tsx

```diff
@@ -92,17 +31,18 @@ interface ExposureHistoryProps {
 const ExposureHistoryProvider = ({
   children,
 }: ExposureHistoryProps): JSX.Element => {
-  const [hasBeenExposed, setHasBeenExposed] = useState<boolean>(false);
+  const [exposureHistory, setExposureHistory] = useState<ExposureHistory>(
+    blankHistory(),
+  );
   const [userHasNewExposure, setUserHasNewExposure] = useState<boolean>(true);
-  const toggleHasExposure = () => {
-    setHasBeenExposed(!hasBeenExposed);
-  };
 
   useEffect(() => {
     if (!isGPS) {
-      const subscription = startListening((exposures: Possible[]) => {
-        console.log(exposures);
-      });
+      const subscription = BTNativeModule.subscribeToExposureEvents(
+        (exposureHistory: ExposureHistory) => {
+          setExposureHistory(exposureHistory);
```

Comment:
Could you clarify what the intended abstraction is supposed to be for a subscriber here? Would the subscriber be responsible for keeping track of all exposure history, and sending the full history, vs. emitting a new history case? 

I believe it's the first one, in which case reconciliation of history is delegated to the service?

Reply:
i think the intent is to keep it as simple as possible. if we can keep the signature to something like:

```typescript
(callback: (exposureHistory: ExposureHistory) => void) => { remove: () => void }
```

(or perhaps

```typescript
(callback: (exposureHistory: ExposureHistory) => void) => () => void

// const unsubscribe = subscribeToGPSUpdates((history) => setHistory(history))
// return unsubscribe
``` 
which i believe is the pattern that redux uses under the hood.)

Then the ExposureHistoryContext really doesnt need to care if the data is coming from the gps side or the bt side or the whatever-side

So i suppose the gps service would process the whole history and pass that in every time the history needs to update.  It really depends on how the data is stored and shipped on gps flow, if its more of a "heres a new data point, please append onto the end of the list" , then we have a pretty different situation and would need to think through how to reconcile the differences if we want to be able to keep this context agnostic to the tracing strategy.

My guess is that it would similar to the BT pattern though given the nature of task, we arent doing real time stuff at this level. We are listening for updates from the HA, and when we get one we check the list of data against the locally stored date gps data, if we get any hits we pass it along in a batch. But i dont have too much context on that side so I'd defer to you on if this a viable approach.